lib/timeout_error.go

```diff
@@ -1,18 +1,37 @@
 package lib
 
+import (
+	"fmt"
+	"time"
+)
+
 // TimeoutError is used when somethings timeouts
-type TimeoutError string
+type TimeoutError struct {
+	place string
+	d     time.Duration
+}
 
-// NewTimeoutError returns a new TimeoutError reporting that timeout has happened at the provieded
-// place
-func NewTimeoutError(place string) TimeoutError {
-	return TimeoutError(place)
+// NewTimeoutError returns a new TimeoutError reporting that timeout has happened
+// at the given place and given duration.
+func NewTimeoutError(place string, d time.Duration) TimeoutError {
+	return TimeoutError{place: place, d: d}
 }
 
+// String returns timeout error in human readable format.
 func (t TimeoutError) String() string {
-	return "Timeout during " + (string)(t)
+	s := t.place + " execution timed out"
+	if t.d.Seconds() > 0 {
```

Comment:
Is it possible for the this to not be more than 0 ? If that is true... does that mean that the user setted it to 0 and we should probably still include it in the message. 
IMO this if should just not be here and we should just have the bottom line ... always.

Reply:
@MStoykov 

> Is it possible for the this to not be more than 0 ? If that is true... does that mean that the user setted it to 0

Yes.

> nd we should probably still include it in the message.
IMO this if should just not be here and we should just have the bottom line ... always.

Maybe, but the error message like `after 0 seconds` seems more meaningless and make user more confuse.

Reply:
Yeah but then, they wouldn't know what the timeout is and if they have a very complex script , where timeout comes from a variable they will now need to find out what the value is through some other means. Which gets harder and harder to debug ...
 While we can just log it and if it's `0` than I think it's perfectly fine that we just tell them we timeouted right away ...

Thinking about this some more ... if we want to have better UX: The whole message should be changed in that case to something like `setup can't be ran as it has a timeout of 0 seconds` as there is no way for the setup to *not* timeout if the timeout is 0 seconds. Or maybe 0 seconds should mean that we have no timeout ? cc @na-- @imiric  ?

Reply:
I've always found `timeout=0` to mean "disable the timeout" altogether, or "use the default value", whatever that may be, not that it should timeout right away, so I would find `setup` not running suprising in that case.

I think the current message makes sense, but we should change the behavior if `timeout=0` implicitly skips execution.

Reply:
@MStoykov after experiment more, I see the current behavior still treat `0s` as a valid timeout, so always show the seconds seems good.

Reply:
Can you also open an issue in which we can discuss if the current timeout=0s behaviour should be changed and ... to what :)

Reply:
@MStoykov  filed https://github.com/loadimpact/k6/issues/1174